runtime-modules/bounty/src/lib.rs

```diff
@@ -1118,127 +1377,210 @@ decl_module! {
         /// - db:
         ///    - `O(N)`
         /// # </weight>
-        #[weight = Module::<T>::submit_oracle_judgement_weight(&judgment)]
+        #[weight = Module::<T>::submit_oracle_judgment_weight(&judgment)]
         pub fn submit_oracle_judgment(
             origin,
-            oracle: BountyActor<MemberId<T>>,
             bounty_id: T::BountyId,
             judgment: OracleJudgment<T::EntryId, BalanceOf<T>>
         ) {
-            let bounty_oracle_manager = BountyActorManager::<T>::ensure_bounty_actor_manager(
+            let bounty = Self::ensure_bounty_exists(&bounty_id)?;
+            BountyActorManager::<T>::ensure_bounty_actor_manager(
                 origin,
-                oracle.clone(),
+                bounty.creation_params.oracle.clone(),
             )?;
 
-            let bounty = Self::ensure_bounty_exists(&bounty_id)?;
-
-            bounty_oracle_manager.validate_actor(&bounty.creation_params.oracle)?;
-
             let current_bounty_stage = Self::get_bounty_stage(&bounty);
 
             Self::ensure_bounty_stage(current_bounty_stage, BountyStage::Judgment)?;
 
-            ensure!(bounty.active_work_entry_count != 0, Error::<T>::NoActiveWorkEntries);
+            let account_id = Self::bounty_account_id(bounty_id);
 
             Self::validate_judgment(&bounty, &judgment)?;
 
             // Lookup for any winners in the judgment.
             let successful_bounty = Self::judgment_has_winners(&judgment);
+            let oracle_reward = bounty.creation_params.oracle_reward;
+            let cherry = bounty.creation_params.cherry;
 
             //
             // == MUTATION SAFE ==
             //
 
             // Return a cherry to a creator.
             if successful_bounty {
+
+                let amount = oracle_reward.saturating_add(cherry).saturating_add(bounty.total_funding);
```

Comment:
Remove solvency checks in both parts of this conditions, and you are also doing them after the mutation safe barrier.